### ISSUE TITLE
Broken PSRCHIVE API when loading archives

### ISSUE THREAD

**Issue by bwmeyers** [v-morello/clfd#3]
Testing in a docker container with PSRCHIVE installed within a Python 3.12 environment (version info: `PSRCHIVE 2024-09-15 (2024-11-14 19546a014)` with shared and static libraries compiled and install successfully). The method for loading archives appears to have changed from
```
import psrchive
a = psrchive.Archive_load('archive')
```

to (an admittedly more Pythonic)

```
import psrchive
a = psrchive.Archive.load('archive')
```

which now means CLFD is version gated to a particular PSRCHIVE commit where this change was introduced. I'm not entirely sure when this happened. 

I think the simple fix is changing this line (or using a try-catch statement to attempt both loading styles before failing):
https://github.com/v-morello/clfd/blob/cf3dd7f058eca038672cea62fc77a397d05feeff/clfd/core.py#L115

**Comment by v-morello** [comment]
Hi @bwmeyers, annoying indeed, your proposed solution seems good. Would you be able to put in an MR for this?

**Comment by v-morello** [comment]
A quick grep reveals that the statement `Archive_load` is used in two places in the code, so maybe a utility function that handles loading archives for both APIs would be justified. Rather than try/except, you could also do something along the lines of:

```python
import psrchive

if hasattr(psrchive, "Archive_load"):
    # ...
else:
    # ...
```

**Comment by bwmeyers** [comment]
Hi @v-morello, yes I can fork and make a PR, no worries. Will do that ASAP.

**Comment by v-morello** [comment]
Solved by PR #4 , thanks! As I said on the PR, I'll make a new release when I can find some decent chunk of time.
